### PR TITLE
Deprecate Rosters, remove from Match, MatchProfiles

### PR DESCRIPTION
**Pull request by sawagh**

### CODE REVIEW

File: api/messages.proto [hunk 1]
```diff
@@ -164,16 +152,8 @@ message MatchProfile {
...
   // Set of pools to be queried when generating a match for this MatchProfile.
-  // The pool names can be used in empty Rosters to specify composition of a
-  // match.
   repeated Pool pools = 3;
 
-  // Set of Rosters for this match request. Could be empty Rosters used to
-  // indicate the composition of the generated Match or they could be partially
-  // pre-populated Ticket list to be used in scenarios such as backfill / join
-  // in progress.
-  repeated Roster rosters = 4;
```
Comment (Laremere) [inline api/messages.proto]: Add to reserved.

File: api/messages.proto [hunk 2]
```diff
@@ -201,9 +181,6 @@ message Match {
   // Tickets belonging to this match.
   repeated Ticket tickets = 4;
 
-  // Set of Rosters that comprise this Match
-  repeated Roster rosters = 5;
```
Comment (Laremere) [inline api/messages.proto]: Add to reserved.

File: api/messages.proto (hunk 1 above)
Comment (sawagh) [reply api/messages.proto]: Done.

File: api/messages.proto (hunk 2 above)
Comment (sawagh) [reply api/messages.proto]: Done

**Pull request merged**